api/agent/agent.go

```diff
@@ -1191,6 +1157,56 @@ type container struct {
 	stats  *drivers.Stats
 }
 
+// IOFS object encapsulates a temporary path and the method to clean it up
+type IOFS interface {
+	Path() string
+	Close() error
+}
+
+type directoryIOFS struct {
+	path string
+}
+
+func (d *directoryIOFS) Path() string {
+	return d.path
+}
+
+func (d *directoryIOFS) Close() error {
+	err := os.RemoveAll(d.path)
+	if err != nil {
+		logrus.WithError(err).Error("Error removing iofs")
+		return err
+	}
+	return nil
+}
+
+func newDirectoryIOFS(cfg *Config) (*directoryIOFS, error) {
+	// XXX(reed): need to ensure these are cleaned up if any of these ops in here fail...
```

Comment:
this looks safe now, can also remove the IOFSOpts option since whoever implements this interface can configure that themselves (it was for mount, which is gone)